inspirehep/modules/workflows/utils/__init__.py

```diff
@@ -158,6 +158,35 @@ def _try_to_log(logfn, *args, **kwargs):
     return _decorator
 
 
+def do_not_repeat(step_id):
+    """Decorator used to skip workflow steps when a workflow is re-run.
+
+    Will store the result of running the workflow step in source_data.persistent_data
+    after running the first time, and skip the step on the following runs.
+    """
+    def decorator(func):
+        @wraps(func)
+        def _do_not_repeat(obj, eng):
+            source_data = obj.extra_data['source_data']
+            if step_id in obj.extra_data['source_data'].setdefault('persistent_data', {}):
+                extra_data_update = source_data['persistent_data'].get(step_id, {})
```

Comment:
you cannot be inside this branch of the `if` if `step_id` is not there, so you don't need to do a `.get` but can use `[step_id]`.